battles/views.py

```diff
@@ -48,13 +49,17 @@ def get_success_message(self, cleaned_data):
         )
 
 
-class BattlesListView(ListView):
+class BattlesListView(LoginRequiredMixin, ListView):
     template_name = 'battles/list_battle.html'
     model = Battle
 
     def get_context_data(self, **kwargs):   # pylint: disable=arguments-differ
         context = super().get_context_data(**kwargs)
         context['on_going'] = Battle.objects.filter(status='ON_GOING')
         context['settled'] = Battle.objects.filter(status='SETTLED')
+        context['has_battle_before'] = Battle.objects.filter(
+            Q(trainer_creator=self.request.user) | Q(trainer_opponent=self.request.user)
+        ).exists()
+        context['trainer_team'] = TrainerTeam.objects.filter(trainer=self.request.user)
```

Comment:
I particularly prefer to have views agnostic to query logic. Instead of writing the queries here, you could either have helpers or write a custom `Queryset` class with methods that filter specific which you would only call here. An example is:

```python
 class CollaboratorQuerySet(models.QuerySet):
    def annotate_collaborator_name(self, name):
        return self.annotate(name=name)
```
and then you would put this on the model:
 
```
objects = models.Manager.from_queryset(CollaboratorQuerySet)()
```